### ISSUE TITLE
Can't start or stop, not even the `lando` command works

### ISSUE THREAD

**Issue by maaxxicarvalho** [lando/lando#608]
Bug Report
==========

## Environment

*Macos version:* 10.13.2
*Docker version:* 17.12.0-ce-rc4-mac44 (21438) - edge channel
*Lando Version:* v3.0.0-beta.31

## `.lando.yml`:*

```yml
name: magento

proxy:
  nginx:
    - local.magento.com

services:
  appserver:
    type: php:7.0
    via: nginx
    ssl: true
    webroot: .
    xdebug: true
    build:
      - "curl -s -O https://files.magerun.net/n98-magerun.phar && chmod +x ./n98-magerun.phar && mv ./n98-magerun.phar /usr/local/bin/magerun"

  database:
    type: mysql:5.7
    portforward: 3306
    creds:
      user: lamp
      password: lamp
      database: lamp

tooling:
  magerun:
    service: appserver
    user: www-data
    cmd: magerun

  composer:
    service: appserver
    description: Run composer commands
    cmd:
      - composer
      - --ansi

  php:
      service: appserver

  mysql:
    user: root
    service: database
    description: Drop into a MySQL shell
```

## Commands that caused the bug:

```bash
lando
```

## Error message:

```bash
warn:  {"message":"API rate limit exceeded for 213.94.140.222. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://developer.github.com/v3/#rate-limiting"}
    (No stack trace)
From previous event:
    at Object.__dirname.setupRoutes.Object.keys.forEach.self.(anonymous function).(anonymous function) [as getReleases] (/snapshot/cli/node_modules/github/lib/index.js:275:40)
    at /snapshot/cli/lib/update.js:0:0
From previous event:
    at AsyncEvents.<anonymous> (/snapshot/cli/lib/update.js:0:0)
    at AsyncEvents.handle (/snapshot/cli/lib/events.js:0:0)
    at /snapshot/cli/lib/events.js:0:0
    at runCallback (timers.js:800:20)
    at tryOnImmediate (timers.js:762:5)
    at processImmediate [as _immediateCallback] (timers.js:733:5)
From previous event:
    at AsyncEvents.emit (/snapshot/cli/lib/events.js:0:0)
    at /snapshot/cli/lib/bootstrap.js:0:0
    at bitmask (/snapshot/cli/node_modules/lodash/lodash.js:10067:25)
    at Object.<anonymous> (/snapshot/cli/bin/lando.js:0:0)
    at Module._compile (pkg/prelude/bootstrap.js:1226:22)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Function.Module.runMain (pkg/prelude/bootstrap.js:1281:12)
    at startup (bootstrap_node.js:200:16)
    at bootstrap_node.js:617:3
error:  TypeError: Cannot read property 'version' of undefined
    at /snapshot/cli/lib/update.js:0:0
From previous event:
    at AsyncEvents.<anonymous> (/snapshot/cli/lib/update.js:0:0)
    at AsyncEvents.handle (/snapshot/cli/lib/events.js:0:0)
    at /snapshot/cli/lib/events.js:0:0
    at runCallback (timers.js:800:20)
    at tryOnImmediate (timers.js:762:5)
    at processImmediate [as _immediateCallback] (timers.js:733:5)
From previous event:
    at AsyncEvents.emit (/snapshot/cli/lib/events.js:0:0)
    at /snapshot/cli/lib/bootstrap.js:0:0
    at bitmask (/snapshot/cli/node_modules/lodash/lodash.js:10067:25)
    at Object.<anonymous> (/snapshot/cli/bin/lando.js:0:0)
    at Module._compile (pkg/prelude/bootstrap.js:1226:22)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Function.Module.runMain (pkg/prelude/bootstrap.js:1281:12)
    at startup (bootstrap_node.js:200:16)
    at bootstrap_node.js:617:3
```

**Comment by serundeputy** [comment]
@maaxxicarvalho thanks for the report.

Looks like you've exceeded the github rate limit:

> "message":"API rate limit exceeded for 213.94.140.222.

Does the `magerun` make calls to github api?  You might try commenting out the `build` step and run `lando start`

**Comment by maaxxicarvalho** [comment]
Hi @serundeputy 
I already tried that, I even tried removing and reinstalling the lando package. None of the commands is working, after the `warn` message it throws an error:

```bash
error:  TypeError: Cannot read property 'version' of undefined
    at /snapshot/cli/lib/update.js:0:0
From previous event:
    at AsyncEvents.<anonymous> (/snapshot/cli/lib/update.js:0:0)
    at AsyncEvents.handle (/snapshot/cli/lib/events.js:0:0)
    at /snapshot/cli/lib/events.js:0:0
    at runCallback (timers.js:800:20)
    at tryOnImmediate (timers.js:762:5)
    at processImmediate [as _immediateCallback] (timers.js:733:5)
From previous event:
    at AsyncEvents.emit (/snapshot/cli/lib/events.js:0:0)
    at /snapshot/cli/lib/bootstrap.js:0:0
    at bitmask (/snapshot/cli/node_modules/lodash/lodash.js:10067:25)
    at Object.<anonymous> (/snapshot/cli/bin/lando.js:0:0)
    at Module._compile (pkg/prelude/bootstrap.js:1226:22)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Function.Module.runMain (pkg/prelude/bootstrap.js:1281:12)
    at startup (bootstrap_node.js:200:16)
    at bootstrap_node.js:617:3
```

**Comment by serundeputy** [comment]
@maaxxicarvalho That usually happens when there is a problem with the `.lando.yml` syntax.

Maybe try a very simple `.lando.yml` file to isolate the problem. Maybe:

```yml
name: myapp
recipe: lamp
```

and see if you can `lando start` that simple file.

**Comment by maaxxicarvalho** [comment]
Hello again @serundeputy 

Apparently, there was some cached configuration under the  `~/.lando/` directory.
After cleaning up the folder I could run the `lando` commands again. Jus to play safe I completely removed the lando application, deleted the `~/.lando` dir and re-installed.

Since we're talking about this, is there any way to flush the lando config cache?

Thanks.

**Comment by serundeputy** [comment]
for a particular app you can do `lando destroy -y`; that will take down the persistent data store too if you have a DB container on the app; fyi.

https://docs.devwithlando.io/cli/destroy.html

Glad it is going for you now! I'll close this reopen if you find a lando problem.

**Comment by kevcom** [comment]
i can confirm that @maaxxicarvalho's solution did work.

to reiterate:

1) Uninstall Lando
2) remove the `~/.lando` folder and everything in it
3) Reinstall Lando

**Comment by mmwebaze** [comment]
I am getting the same error message and none of the solutions provided in this thread work. I haven't tried @kevcom's though.

**Comment by rbirene** [comment]
I had a similar approach to others when I got an issue where some of the Docker containers weren't shutting down and I couldn't run the `lando stop` command or even uninstall Lando because of it.

**Docker**
- Delete any containers
- Force quit Docker Desktop
- Remove from Applications
- Delete the `~/.docker` folder and everything in it

**Lando**
- Run the uninstall command ([https://docs.lando.dev/getting-started/uninstalling.html](https://docs.lando.dev/getting-started/uninstalling.html))
- Delete the `~/.lando` folder and everything in it
- Reinstall Lando

When I ran lando start for my app again, everything seemed to be working fine and the containers that were previously failing seemed to be spinning up normally again.

Hope this helps anyone else!
